pkg/local_object_storage/metabase/v2/put.go

```diff
@@ -305,3 +335,87 @@ func updateBlobovniczaID(tx *bbolt.Tx, addr *objectSDK.Address, id *blobovnicza.
 
 	return bkt.Put(objectKey, *id)
 }
+
+// updateSpliInfo for existing objects if storage filled with extra information
+// about last object in split hierarchy or linking object.
+func updateSplitInfo(tx *bbolt.Tx, addr *objectSDK.Address, from *objectSDK.SplitInfo) error {
+	bkt := tx.Bucket(rootBucketName(addr.ContainerID()))
+	if bkt == nil {
+		// if object doesn't exists and we want to update split info on it
+		// then ignore, this should never happen
+		return ErrIncorrectSplitInfoUpdate
+	}
+
+	objectKey := objectKey(addr.ObjectID())
+
+	rawSplitInfo := bkt.Get(objectKey)
+	if len(rawSplitInfo) == 0 {
+		return ErrIncorrectSplitInfoUpdate
+	}
+
+	to := objectSDK.NewSplitInfo()
+
+	err := to.Unmarshal(rawSplitInfo)
+	if err != nil {
+		return fmt.Errorf("can't unmarshal split info from root index: %w", err)
+	}
+
+	result := mergeSplitInfo(from, to)
+
+	rawSplitInfo, err = result.Marshal()
+	if err != nil {
+		return fmt.Errorf("can't marhsal merged split info: %w", err)
+	}
+
+	return bkt.Put(objectKey, rawSplitInfo)
+}
+
+// splitInfoFromObject returns split info based on last or linkin object.
+// Otherwise returns nil, nil.
+func splitInfoFromObject(obj *object.Object) (*objectSDK.SplitInfo, error) {
+	if obj.Parent() == nil {
+		return nil, nil
+	}
+
+	info := objectSDK.NewSplitInfo()
+	info.SetSplitID(obj.SplitID())
+
+	switch {
+	case isLinkObject(obj):
+		info.SetLink(obj.ID())
+	case isLastObject(obj):
+		info.SetLastPart(obj.ID())
+	default:
+		return nil, ErrIncorrectRootObject // should never happen
+	}
+
+	return info, nil
+}
+
+// mergeSplitInfo ignores conflicts and rewrites `to` with non empty values
+// from `from`.
+func mergeSplitInfo(from, to *objectSDK.SplitInfo) *objectSDK.SplitInfo {
```

Comment:
This function can return no result since `from` is a pointer. I am also developing a code that implements exactly the same function , so I propose to share it in a common package (not in this PR).